<!--
LiDARドライバとAPI送信ノードを同時起動
起動コマンド:roslaunch lightrover_ros lidar_with_api.launch
-->

<launch>
  <!-- TF設定 -->
  <node pkg="tf" type="static_transform_publisher" name="odom_2_base_link" 
        args="-0.042 0 0.1094 -1.5708 0 0 base_link lidar_link 1" /> 

  <!-- YDLiDAR ドライバノード -->
  <node name="ydlidar_ros_driver_node" pkg="ydlidar_ros_driver" type="ydlidar_ros_driver_node" 
        output="screen" respawn="false">
    <!-- string property -->
    <param name="port" type="string" value="/dev/ydlidar"/>  
    <param name="frame_id" type="string" value="lidar_link"/>
    <param name="ignore_array" type="string" value=""/>

    <!-- int property -->
    <param name="baudrate" type="int" value="115200"/>  
    <param name="lidar_type" type="int" value="1"/>  
    <param name="device_type" type="int" value="0"/>  
    <param name="sample_rate" type="int" value="3"/>  
    <param name="abnormal_check_count" type="int" value="4"/>  

    <!-- bool property -->
    <param name="resolution_fixed" type="bool" value="true"/>
    <param name="auto_reconnect" type="bool" value="true"/>
    <param name="reversion" type="bool" value="false"/>
    <param name="inverted" type="bool" value="true"/>
    <param name="isSingleChannel" type="bool" value="true"/>
    <param name="intensity" type="bool" value="false"/>
    <param name="support_motor_dtr" type="bool" value="true"/>
    <param name="invalid_range_is_inf" type="bool" value="false"/>
    <param name="point_cloud_preservative" type="bool" value="false"/>

    <!-- float property -->
    <param name="angle_min" type="double" value="-180" />
    <param name="angle_max" type="double" value="180" />
    <param name="range_min" type="double" value="0.1" />
    <param name="range_max" type="double" value="12.0" />
    <param name="frequency" type="double" value="10.0"/>
  </node>

  <!-- LiDAR API Publisher ノード -->
  <node pkg="lightrover_ros" type="lidar_api_publisher.py" name="lidar_api_publisher" output="screen">
    <!-- 設定ファイルからパラメータを読み込む -->
    <rosparam command="load" file="$(find lightrover_ros)/configuration_files/config.yaml" />
  </node>

  <!-- RViz（可視化用、必要に応じてコメントアウト） -->
  <!-- <node pkg="rviz" type="rviz" args="-d '$(find lightrover_ros)/configuration_files/rviz/gmapping.rviz'" 
        name="visualizer" respawn="true" /> -->
</launch>
